src/app/services/ros.service.ts

```diff
@@ -84,12 +94,21 @@ export class RosService {
       callback(silColorHex);
     });
   }
+
   // Method to subscribe to gps topic
   subscribeToGPS(callback: (data: any) => void) {
     this.gpsSubscriber.subscribe((message: any) => {
       const coordinate = {lat : message.latitude, lng: message.longitude}
       callback(coordinate)
     });
   }
+
+  // Method to subscribe to ArUco markers
+  subscribeToArucoMarkers(callback: (data: ArucoMarker) => void) {
```

Comment:
out of interest, why is the aruco marker subscriber not subscribed to the topic in its constructor?